pennylane/ops/qubit/hamiltonian.py

```diff
@@ -616,13 +616,30 @@ def compare(self, other):
         if isinstance(other, Hamiltonian):
             self.simplify()
             other.simplify()
-            return self._obs_data() == other._obs_data()  # pylint: disable=protected-access
+            if len(self._obs_data()) != len(other._obs_data()):
+                return False
+            print(self._obs_data(), " and ", other._obs_data())
+            for self_data in self._obs_data():  # pylint: disable=protected-access
```

Comment:
You can just store `self._obs_data()` and `other._obs_data()` in variables before you start the loop. That way you don't need to call `_obs_data()` for every iteration of the loop.